### DISCUSSION TITLE
Documentation questions on irbempy and coordinates

### DISCUSSION THREAD

**Discussion by jvrod23** [spacepy/spacepy#492] [Support and Q&A]
I have several questions (and possibly one minor bug) regarding the documentation of the irbemby and coordinates librarites.  To give some context: I'm performing magnetic field line tracing using the POES satellite locations as inputs.  These locations are provided in geodetic coordinates (satellite altitude, lat, lon).

1. It would be good to provide a link to 'find_footprint' from https://spacepy.github.io/irbempy.html.  I've been using the documentation in the code itself.
2. Is the 'alt' input parameter to 'find_footprint' in GDZ or GEO coordinates?  In other words, is this the altitude above the WGS84 spheroid or above some spherical geocentric surface (e.g., with a 6371-km radius)?
3. It would be helpful to note that in GDZ coordinates, the first element is the altitude in km, not the radius in Re.  Steve clarified this in https://github.com/spacepy/spacepy/issues/213.
4. I believe the conversion from GDZ to GEO does not return the correct units string.  Here, a is an (n,3) ndarray created from the POES satellite location contained in the data files.

from spacepy import coordinates as coord
sem2_loci = coord.Coords(a, 'GDZ', 'sph')
geog_coord = sem2_loci.convert('GEO', 'sph')

geog_coord.radi appears to be in units of Re (appropriately), but geog_coord.units is ['km', 'deg', 'deg']

5. Into which coordinate system do 'get_Lstar' and 'find_footprint' convert the 'loci' input before proceeding with the model calls and field line tracing?  I am hoping that if they convert to the same coordinate system, I could do this beforehand so that this conversion is not duplicated within 'get_Lstar' and 'find_footprint'.

Thanks!

**Comment by jtniehof** [comment]
Item 1 the same sort of thing as #489 and I can fix it the same way; PR shortly (EDIT: #493, but won't show up on web until next release.)

Item 4 likely relates to #477 and I've noted this conversation there.

The rest I'll leave to @drsteve.
(Edit by @drsteve to add)
1. As @jtniehof noted, making sure the function appears in the built docs is definitely something that needs to happen. It looks like it should be fixed by #493 and then the online docs will update to include it when we push the next release. We're aiming to get that out fairly soon.
2. Footpoint altitude is in geodetic coordinates. I.e., altitude above WGS84 ellipsoid. That should be made explicit it the docs. I can add that in the general tidy I've been doing as part of #296
3. I've been updating the documentation for `spacepy.coordinates` as part of #296 and that's taken care of there. The short story is that the assumed distance units are Re everywhere except GDZ, where the distance is in km.
4. Taken care of in #296. That was an oversight in how the unit handling was originally done on our end. Essentially, we were preserving the units across transformations and converting between spherical and Cartesian, but the Re<->km whenever geodetic was involved wasn't being handled.
5. They'll do least transformation if you hand in Cartesian GEO coordinates.

**Reply by drsteve** [reply]
Comments added to the above answer

**Reply by jvrod23** [reply]
Thanks, very helpful feedback!

**Reply by jtniehof** [reply]
Should we document cartesian GEO as the "least transformation" option somewhere?

Also, given that the Q&A discussions are supposed to be sort of Stack Overflow-y, I can edit your comments into my reply, but I'm also curious if you can edit them in to my reply.

**Reply by drsteve** [reply]
Looks like yes on being able to edit your reply. I've moved my comments up.

Should we document GEO as "least transformation"? Sure, although I'm not sure how much it matters, performance-wise. The IRBEM routines under `find_footpoint` do a lot of coordinate transformations internally, and relative to the L* calculations a few coordinate transforms are insignificant.
